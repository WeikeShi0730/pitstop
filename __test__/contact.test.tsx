import {
  render,
  RenderResult,
  screen,
  fireEvent,
} from "@testing-library/react";
import Contact from "../components/contact.component";

let documentBody: RenderResult;
describe("<Contact />", () => {
  beforeEach(() => (documentBody = render(<Contact />)));
  it("shows basic fields in <Contact />", () => {
    expect(
      documentBody.getByRole("heading", { name: /Contact us/i })
    ).toBeInTheDocument();
    expect(
      documentBody.getByRole("textbox", { name: /name/i })
    ).toBeInTheDocument();
    expect(
      documentBody.getByRole("textbox", { name: /email/i })
    ).toBeInTheDocument();
    expect(
      documentBody.getByRole("textbox", { name: /message/i })
    ).toBeInTheDocument();
    expect(
      documentBody.getByRole("button", { name: /submit/i })
    ).toBeInTheDocument();

    const { baseElement } = documentBody;
    expect(baseElement).toMatchSnapshot();
  });
  it("Name field in <Contact />", () => {
    fireEvent.change(documentBody.getByRole("textbox", { name: /name/i }), {
      target: {
        value: "Test Name",
      },
    });
    expect(
      (
        documentBody.getByRole("textbox", {
          name: /name/i,
        }) as HTMLInputElement
      ).value
    ).toBe("Test Name");
  });
  it("Email field in <Contact />", () => {
    fireEvent.change(documentBody.getByRole("textbox", { name: /email/i }), {
      target: {
        value: "Test Email",
      },
    });
    expect(
      (
        documentBody.getByRole("textbox", {
          name: /email/i,
        }) as HTMLInputElement
      ).value
    ).toBe("Test Email");
  });
  it("Message field in <Contact />", () => {
    fireEvent.change(documentBody.getByRole("textbox", { name: /message/i }), {
      target: {
        value: "Test Message",
      },
    });
    expect(
      (
        documentBody.getByRole("textbox", {
          name: /message/i,
        }) as HTMLInputElement
      ).value
    ).toBe("Test Message");
    // fireEvent.submit(documentBody.getByRole("button", { name: /submit/i }));
    // expect(onSubmit).toBeCalled();
  });
});
