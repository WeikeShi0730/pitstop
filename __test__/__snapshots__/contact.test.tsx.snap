// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Contact /> shows basic fields in <Contact /> 1`] = `
<body>
  <div>
    <div
      class="min-h-content flex justify-center items-center text-sm md:text-base"
    >
      <div
        class="max-w-xs sm:max-w-xl md:max-w-2xl lg:max-w-3xl m-auto my-10 text-slate-700 bg-slate-100 bg-opacity-30 rounded-lg py-8 px-10 shadow-md hover:shadow-slate-500 transition-all ease-in-out duration-200 font-light"
      >
        <form>
          <h2
            class="text-center text-lg md:text-2xl text-orange-theme mb-4"
          >
            Contact us
          </h2>
          <p
            class="text-center text-slate-700 sm:w-2/3 w-full mx-auto mb-8"
          >
            We want to hear from you! Please feel free to use the form below to speak your mind, and we actaully reply 
            <span>
              ðŸ˜ƒ
            </span>
          </p>
          <div
            class="relative mb-4 sm:w-2/3 w-full m-auto"
          >
            <label
              class="relative text-sm md:text-base"
              for="name"
            >
              <input
                autocomplete="off"
                class="w-full p-2 border-b border-slate-700 outline-none bg-transparent mb-4 rounded-none transition duration-200"
                id="name"
                name="name"
                placeholder=" "
                required=""
                type="text"
              />
              <span
                class="absolute left-0 top-0 px-2 transition duration-200 input-text"
              >
                Name
              </span>
            </label>
          </div>
          <div
            class="relative mb-4 sm:w-2/3 w-full m-auto"
          >
            <label
              class="relative text-sm md:text-base"
              for="email"
            >
              <input
                class="w-full p-2 border-b border-slate-700 outline-none bg-transparent mb-4 rounded-none transition duration-200"
                id="email"
                name="email"
                placeholder=" "
                required=""
                type="email"
              />
              <span
                class="absolute left-0 top-0 px-2 transition duration-200 input-text"
              >
                Email
              </span>
            </label>
          </div>
          <div
            class="relative mb-4 sm:w-2/3 w-full m-auto"
          >
            <label
              class="text-sm md:text-base"
              for="message"
            >
              Message
            </label>
            <textarea
              class="text-sm md:text-base w-full h-auto p-2 border border-slate-700 outline-none bg-transparent my-4 rounded-lg"
              id="message"
              name="message"
              required=""
              rows="4"
            />
          </div>
          <div
            class="flex justify-center items-center mt-4"
          >
            <button
              class="text-sm md:text-base bg-slate-700 py-2 px-4 text-slate-100 rounded outline-none font-light hover:shadow-md hover:shadow-slate-700 hover:bg-slate-500 hover:text-white transition-all ease-in-out duration-200"
              type="submit"
            >
              submit
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
</body>
`;
